### PR TITLE
fix: automation being triggered but not polled

### PR DESCRIPTION
**Pull request by rileymd88**
This fixes an issue where a user may be able to trigger an automation from the button but does not have permission to poll for its status. For this to happen the following needs to be true:

Run mode: triggered needs to be checked in the button properties
The user that clicks the button is not the owner of the automation
Start block Run Mode in the automation needs to have Run asynchronously set to true

### CODE REVIEW

File: src/utils/automation-helper.js [hunk 1]
```diff
@@ -32,6 +32,9 @@ export const getAutomation = async (automationId) => {
 
 export const getAutomationRun = async (automationId, runId) => {
   const response = await fetch(`../api/v1/automations/${automationId}/runs/${runId}`);
+  if (response.status !== 200) {
```
Comment (cbt1) [inline src/utils/automation-helper.js:35]: What kind of response code (status) is returned when the user does not have permissions? If 403, you should consider to only check for that

File: src/utils/automation-helper.js [hunk 2]
```diff
@@ -32,6 +32,9 @@ export const getAutomation = async (automationId) => {
...
+    throw Error;
```
Comment (cbt1) [inline src/utils/automation-helper.js:36]: You are throwing the `Error` function here. Sort of like `throw () => {}`.

Consider something like `throw new Error()` instead.

Comment (rileymd88) [reply src/utils/automation-helper.js]: Fixed this in the latest commit

**Review by cbt1 — APPROVED** [review]

**Comment by cbt1** [comment]
Fix the failing unit tests and you are good to go

**Comment by rileymd88** [comment]
> Fix the failing unit tests and you are good to go

Done in the latest commit :)

**Comment by niekvanstaveren** [comment]
If still valid this needs to be recreated from a direct branch, not from a Fork

**Pull request closed without merging**